api/src/main/java/org/openmrs/Allergy.java

```diff
@@ -247,50 +247,64 @@ public Date getDateLastUpdated() {
+				&& hasAllergenWithSameNonCodedAllergen(allergy) && hasSameSeverity(allergy) && hasSameComment(allergy)
+				&& hasSameReactions(allergy);
+	}
+	
+	
+	
+	private boolean hasSameAllergyId(Allergy allergy) {
+		return OpenmrsUtil.nullSafeEquals(getAllergyId(), allergy.getAllergyId());
+	}
+	
+	private boolean hasSamePatient(Allergy allergy) {
 		if (!OpenmrsUtil.nullSafeEquals(getPatient(), allergy.getPatient())) {
 			//if object instances are different but with the same patient id, then not changed
 			if (getPatient() != null && allergy.getPatient() != null) {
-				if (!OpenmrsUtil.nullSafeEquals(getPatient().getPatientId(), allergy.getPatient().getPatientId())) {
-					return false;
-				}
-			}
-			else {
+				return OpenmrsUtil.nullSafeEquals(getPatient().getPatientId(), allergy.getPatient().getPatientId());
+			} else {
 				return false;
 			}
 		}
+		return true;
+	}
+	
+	
+	private boolean hasAllergenWithSameCodedAllergen(Allergy allergy) {
 		if (!OpenmrsUtil.nullSafeEquals(getAllergen().getCodedAllergen(), allergy.getAllergen().getCodedAllergen())) {
 			//if object instances are different but with the same concept id, then not changed
 			if (getAllergen().getCodedAllergen() != null && allergy.getAllergen().getCodedAllergen() != null) {
-				if (!OpenmrsUtil.nullSafeEquals(getAllergen().getCodedAllergen().getConceptId(), allergy.getAllergen().getCodedAllergen().getConceptId())) {
-					return false;
-				}
-			}
-			else {
+				return OpenmrsUtil.nullSafeEquals(getAllergen().getCodedAllergen().getConceptId(),
+					allergy.getAllergen().getCodedAllergen().getConceptId());
+			} else {
 				return false;
 			}
 		}
-		if (!OpenmrsUtil.nullSafeEquals(getAllergen().getNonCodedAllergen(), allergy.getAllergen().getNonCodedAllergen())) {
-			return false;
-		}
+		return true;
+	}
+	
+	private boolean hasAllergenWithSameNonCodedAllergen(Allergy allergy) {
+		return OpenmrsUtil.nullSafeEquals(getAllergen().getNonCodedAllergen(), allergy.getAllergen().getNonCodedAllergen());
+	}
+	
+	
+	private boolean hasSameSeverity(Allergy allergy) {
 		if (!OpenmrsUtil.nullSafeEquals(getSeverity(), allergy.getSeverity())) {
 			//if object instances are different but with the same concept id, then not changed
-			if (getSeverity() != null && allergy.getSeverity() != null) {
-				if (!OpenmrsUtil.nullSafeEquals(getSeverity().getConceptId(), allergy.getSeverity().getConceptId())) {
-					return false;
-				}
-			}
-			else {
+			if (getSeverity()!= null && allergy.getSeverity() != null) {
+				return OpenmrsUtil.nullSafeEquals(getSeverity().getConceptId(), allergy.getSeverity().getConceptId());
+			} else {
 				return false;
 			}
 		}
-		if (!OpenmrsUtil.nullSafeEquals(getComment(), allergy.getComment())) {
-			return false;
-		}
-		return hasSameReactions(allergy);
+		return true;
 	}
+
+	private boolean hasSameComment(Allergy allergy) {
+		return OpenmrsUtil.nullSafeEquals(getComments(), allergy.getComment());
```

Comment:
Shouldn't this be:

```suggestion
		return OpenmrsUtil.nullSafeEquals(getComments(), allergy.getComments());
```

or

```suggestion
		return OpenmrsUtil.nullSafeEquals(getComment(), allergy.getComment());
```
?